documentation/book/con-kafka-listeners.adoc

```diff
@@ -108,22 +138,64 @@ listeners:
         nodePort: 32100
       brokers:
       - broker: 0
-        advertisedHost: example.host.1
-        advertisedPort: 12340
         nodePort: 32000
       - broker: 1
-        advertisedHost: example.host.2
-        advertisedPort: 12341
         nodePort: 32001
       - broker: 2
-        advertisedHost: example.host.3
-        advertisedPort: 12342
         nodePort: 32002
 # ...
 ----
 
 For more information on using node ports to access Kafka, see xref:proc-accessing-kafka-using-nodeports-{context}[].
 
+.Customising advertised addresses on external listeners
+
+By default, {ProductName} will try to automatically determine the hostnames and ports which the Kafka cluster should advertise to its clients.
+In some situations this might not be sufficient, because the infrastructure where you run {ProductName} might not provide the right hostname on port under which Kafka can be accessed.
+You can customize the advertised hostname and port in the `overrides` property of the external listener.
+{ProductName} will automatically configure the advertised address in the Kafka brokers and add it to the broker certificates, so that it can be used for TLS hostname verification.
+
+.An example where the external listener is configured with overrides for advertised addresses
+[source,yaml,subs="attributes+"]
+----
+# ...
+listeners:
+  external:
+    type: routes
+    authentication:
+      type: tls
+    overrides:
+      brokers:
+      - broker: 0
+        advertisedHost: example.hostname.0
+        advertisedPort: 12340
+      - broker: 1
+        advertisedHost: example.hostname.1
+        advertisedPort: 12341
+      - broker: 2
+        advertisedHost: example.hostname.2
+        advertisedPort: 12342
+# ...
+----
+
+Additionally, you can specify the name of the bootstrap service.
+This name will be added to the broker certificates and can be used for TLS hostname verification.
```

Comment:
because most of the above examples are with `routes` we could say that the `bootstrap.address` the customisation is possible for all the external listener types.